### PR TITLE
Address CI failures due to the removal of the positive filter by excluding failing and long tests

### PR DESCRIPTION
**Pull request by aalabsi-amd**
## Motivation

The work in this PR https://github.com/ROCm/TheRock/pull/2888 added more tests to the CI for MIOpen and some started to fail for windows gfx1151, they are added to the windows negative filter in this PR.
Also some occasional timeouts due to the increase of number of tests and their run time, added to the negative filter here.

## Technical Details
Collected results from this issue https://github.com/ROCm/rocm-libraries/issues/3956#issuecomment-3775398643
New failing tests added the windows negative filter:
```
# New tests added and failing with the removal of the positive filter
    negative_filter.append("Smoke/GPU_MultiMarginLoss_*")
    negative_filter.append("CPU_TuningPolicy_NONE")
    negative_filter.append("Full/CPU_CandidateSelection_NONE*")
    negative_filter.append("GPU_GroupConv2D_Forward_BFP16*")  # It tries to hipMalloc 2.5GB of memory
    negative_filter.append("GPU_OpTensorFwdBiasTest_FP32*")
    negative_filter.append("GPU_ConvGrpBiasActivInfer3D_BFP16*")
    negative_filter.append("GPU_ConvGrpActivInfer_BFP16*")
```
New long running tests added to the general negative filter:
```
negative_filter.append("Full/GPU_MIOpenDriverConv2dTransTest_BFP16*")  # 1 min 34 sec
negative_filter.append("Smoke/GPU_Reduce_FP32*")  # 1 min 30 sec
negative_filter.append("Smoke/GPU_Reduce_FP16*")  # 1 min 11 sec
```
## Test Plan

The CI need to start passing again specifically windows gfx1151
Check if there is any more test timeout, and continuing collecting run time

## Test Result

TODO: watch the CI

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

File: build_tools/github_actions/test_executable_scripts/test_miopen.py [hunk 1]
```diff
@@ -125,6 +128,19 @@
         "*CPU_UnitTestConvSolverImplicitGemmGroupWrwXdlopsDevApplicability_FP16.ConvHipImplicitGemmGroupWrwXdlops*"
     )
 
+    # New tests added and failing with the removal of the positive filter
```
Comment (ScottTodd) [inline build_tools/github_actions/test_executable_scripts/test_miopen.py:131]: When I test these changes locally on my Windows gfx1151 dev machine, I'm still seeing at least one crash that may be nondeterministic.

Logs near the failure:
```
[----------] 1 test from Smoke/GPU_GemmWrwUniversal_LargeMemory_FP32
[ RUN      ] Smoke/GPU_GemmWrwUniversal_LargeMemory_FP32.GemmWrwUniversal/0
[       OK ] Smoke/GPU_GemmWrwUniversal_LargeMemory_FP32.GemmWrwUniversal/0 (28411 ms)
[----------] 1 test from Smoke/GPU_GemmWrwUniversal_LargeMemory_FP32 (28411 ms total)

[----------] 1 test from Smoke/GPU_GemmWrwUniversal_LargeMemory_FP16
[ RUN      ] Smoke/GPU_GemmWrwUniversal_LargeMemory_FP16.GemmWrwUniversal/0
Exception Code: 0x80000003
 #0 0x00007ffe92d1ea65 amd_comgr_symbolize (D:\installs\21108104773\bin\amd_comgr0702.dll+0x146ea65)
 #1 0x00007fff79011989 (C:\WINDOWS\System32\ucrtbase.dll+0xc1989)
 #2 0x00007fff78ff4ab1 (C:\WINDOWS\System32\ucrtbase.dll+0xa4ab1)
 #3 0x00007fff790119de (C:\WINDOWS\System32\ucrtbase.dll+0xc19de)
 #4 0x00007fff73b81134 (C:\WINDOWS\SYSTEM32\VCRUNTIME140.dll+0x1134)
 #5 0x00007fff73b92d1d (C:\WINDOWS\SYSTEM32\VCRUNTIME140.dll+0x12d1d)
 #6 0x00007fff73b8f47f (C:\WINDOWS\SYSTEM32\VCRUNTIME140.dll+0xf47f)
 #7 0x00007fff7bbe62ff (C:\WINDOWS\SYSTEM32\ntdll.dll+0x1662ff)
 #8 0x00007fff7ba92327 (C:\WINDOWS\SYSTEM32\ntdll.dll+0x12327)
 #9 0x00007fff7ba8a961 (C:\WINDOWS\SYSTEM32\ntdll.dll+0xa961)
#10 0x00007fff7916782a (C:\WINDOWS\System32\KERNELBASE.dll+0xc782a)
#11 0x00007fff73b852c7 (C:\WINDOWS\SYSTEM32\VCRUNTIME140.dll+0x52c7)
#12 0x00007ffe996e7c76 miopen::HIPOCKernel::HIPOCKernel(struct miopen::HIPOCProgram, class std::basic_string<char, struct std::char_traits<char>, class std::allocator<char>>, class std::vector<unsigned __int64, class std::allocator<unsigned __int64>>, class std::vector<unsigned __int64, class std::allocator<unsigned __int64>>) (D:\installs\21108104773\bin\MIOpen.dll+0x127c76)
#13 0x00007ffe9acc1632 miopen::Handle::ReserveExtraStreamsInPool(int) const (D:\installs\21108104773\bin\MIOpen.dll+0x1701632)
#14 0x00007ff781464665 (D:\installs\21108104773\bin\miopen_gtest.exe+0x464665)
```

Seems like since that's an exception and not just a test failure, it brings down the whole test process. Is there extra logging I could enable that would help diagnose? Then maybe we can choose a category of tests to disable instead of going one failing test at a time.

Comment (aalabsi-amd) [reply build_tools/github_actions/test_executable_scripts/test_miopen.py:131]: maybe try with these log options:
`MIOPEN_ENABLE_LOGGING_CMD=1 MIOPEN_ENABLE_LOGGING=1 MIOPEN_LOG_LEVEL=6`
or `AMD_LOG_LEVEL=4 MIOPEN_LOG_LEVEL=7`

Comment (ScottTodd) [reply build_tools/github_actions/test_executable_scripts/test_miopen.py:131]: Got a bit more info for some specific test failures that fail somewhat consistently using those env vars: https://gist.github.com/ScottTodd/db8e9b8ee4a10ddaa237ec1b6262f6a0

```
[----------] 79 tests from Full/GPU_UnitTestConvSolverDirectNaiveBwd_FP16
[ RUN      ] Full/GPU_UnitTestConvSolverDirectNaiveBwd_FP16.ConvDirectNaiveConvBwd/0
MIOpen(HIP): Info [] MIOPEN_FIND_MODE = DYNAMIC_HYBRID(5)
MIOpen(HIP): Info2 [default_allocator] GetAvailableMemory 51564732416
MIOpen(HIP): Info2 [default_allocator] hipMalloc 102760448 at 0000003006000000 Ok
MIOpen(HIP): Info2 [default_allocator] GetAvailableMemory 51359211520
MIOpen(HIP): Info2 [default_allocator] hipMalloc 524288 at 000000300C200000 Ok
MIOpen(HIP): Info2 [default_allocator] GetAvailableMemory 51224993792
MIOpen(HIP): Info2 [default_allocator] hipMalloc 25690112 at 0000003010200000 Ok
MIOpen(HIP): Info [AmdRocmMetadataVersionDetect] ROCm MD version AMDHSA_COv3, HIP version 7.2.53150, MIOpen version 3.5.1.37865a97a6
MIOpen(HIP): Info2 [PrepareInvoker] Preparing kernel: naive_conv_ab_nonpacked_bwd_nchw_half_double_half
MIOpen(HIP): Info2 [SQLiteBase] Initializing system database file ""
MIOpen(HIP): Info [KernDb] database not present
MIOpen(HIP): Info2 [SQLiteBase] Initializing user database file "C:\\Users\\Scott\\.miopen\\cache\\3.5.1.37865a97a6\\gfx1151_20.ukdb"
MIOpen(HIP): Trace [Exec] 24568:PRAGMA journal_mode=WAL;
MIOpen(HIP): Trace [Exec] 24568:CREATE TABLE IF NOT EXISTS `kern_db` (`id` INTEGER PRIMARY KEY ASC,`kernel_name` TEXT NOT NULL,`kernel_args` TEXT NOT NULL,`kernel_blob` BLOB NOT NULL,`kernel_hash` TEXT NOT NULL,`uncompressed_size` INT NOT NULL);CREATE UNIQUE INDEX IF NOT EXISTS `idx_kern_db` ON kern_db(kernel_name, kernel_args);
MIOpen(HIP): Info2 [KernDb] Database created successfully
MIOpen(HIP): Trace [Exec] 24568:PRAGMA table_info(kern_db);
MIOpen(HIP): Info2 [LoadBinary] Loading binary for: "naive_conv.cpp.obj"; args:  -DMIOPEN_USE_FP16=1 -DMIOPEN_USE_FP16x4=0 -DMIOPEN_USE_FP16x8=0 -DMIOPEN_USE_FP32=0 -DMIOPEN_USE_INT8=0 -DMIOPEN_USE_BFP16=0 -DMIOPEN_USE_INT32=0 -DMIOPEN_USE_RNE_BFLOAT16=1 -DMIOPEN_FP8_IEEE_EXPONENT_BIAS=0 -DMIOPEN_FP8_CLIPPING=1 -mcpu=gfx1151
MIOpen(HIP): Info2 [Prepare] SELECT kernel_blob, kernel_hash, uncompressed_size FROM kern_db WHERE (kernel_name = 'naive_conv.cpp.obj') AND (kernel_args = ' -DMIOPEN_USE_FP16=1 -DMIOPEN_USE_FP16x4=0 -DMIOPEN_USE_FP16x8=0 -DMIOPEN_USE_FP32=0 -DMIOPEN_USE_INT8=0 -DMIOPEN_USE_BFP16=0 -DMIOPEN_USE_INT32=0 -DMIOPEN_USE_RNE_BFLOAT16=1 -DMIOPEN_FP8_IEEE_EXPONENT_BIAS=0 -DMIOPEN_FP8_CLIPPING=1 -mcpu=gfx1151');
MIOpen(HIP): Info2 [Measure] Db::FindRecord time: 6.5316 ms
MIOpen(HIP): Info2 [LoadBinary] Successfully loaded binary for: "naive_conv.cpp.obj"; args:  -DMIOPEN_USE_FP16=1 -DMIOPEN_USE_FP16x4=0 -DMIOPEN_USE_FP16x8=0 -DMIOPEN_USE_FP32=0 -DMIOPEN_USE_INT8=0 -DMIOPEN_USE_BFP16=0 -DMIOPEN_USE_INT32=0 -DMIOPEN_USE_RNE_BFLOAT16=1 -DMIOPEN_FP8_IEEE_EXPONENT_BIAS=0 -DMIOPEN_FP8_CLIPPING=1 -mcpu=gfx1151
MIOpen(HIP): Info2 [run] kernel_name = naive_conv_ab_nonpacked_bwd_nchw_half_double_half, global_work_dim = { 67108864, 1, 1 }, local_work_dim = { 256, 1, 1 }
MIOpen(HIP): Error [FreeBuf] hipFree(buf_)failed at line: 87 in file: C:/home/runner/_work/TheRock/TheRock/rocm-libraries/projects/miopen/test/gtest/..\workspace.hpp
MIOpen(HIP): Warning [default_deallocator] hipMemPtrGetInfo at 0000003010200000 status: 719
Traceback (most recent call last):
  File "D:\projects\TheRock\build_tools\github_actions\test_executable_scripts\test_miopen.py", line 184, in <module>
    subprocess.run(cmd, cwd=THEROCK_DIR, check=True, env=environ_vars)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python313\Lib\subprocess.py", line 577, in run
    raise CalledProcessError(retcode, process.args,
                             output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['D:\\installs\\21108104773\\bin\\/miopen_gtest', '--gtest_filter=Full/GPU_UnitTestConvSolver*']' returned non-zero exit status 3.
```

Comment (amd-justchen) [reply build_tools/github_actions/test_executable_scripts/test_miopen.py:131]: Also just to note, I observed another MIOpen test failure in real time, which also corroborates the nondeterministic test failure observation, as well as how it is bringing down the whole test process. In CI, that will hang the runner until the 60 min timeout, despite failing quite early:

>Wed, 21 Jan 2026 03:52:32 GMT [  FAILED  ] ...
>Wed, 21 Jan 2026 03:52:32 GMT [ RUN      ] ...
>Wed, 21 Jan 2026 04:40:34 GMT Error: The action 'Test' has timed out after 60 minutes.

So, I think any timeouts that we see, are actually *also* MIOpen Test crashes, along with ones that do fail, but allows the runner to continue without timeout. It also offers some insight into why our Strix-Halo runner queues are so long since this step is maxing out at 1 hour vs around 20 minutes and making systems offline reducing capacity.

https://github.com/ROCm/rocm-libraries/issues/3956#issuecomment-3772439579 

```haskell

Wed, 21 Jan 2026 03:52:32 GMT C:/home/runner/_work/TheRock/TheRock/rocm-libraries/projects/miopen/test/gtest/tensor_fwd_bias_ocl_hip.cpp(351): error: Value of: error == 0
Wed, 21 Jan 2026 03:52:32 GMT   Actual: false
Wed, 21 Jan 2026 03:52:32 GMT Expected: true
Wed, 21 Jan 2026 03:52:32 GMT GPU outputs do not match each other. Error: 0.030906672816040103
Wed, 21 Jan 2026 03:52:32 GMT [  FAILED  ] Smoke/GPU_OpTensorFwdBiasTest_FP32.PortTest/40, where GetParam() = (96-byte object <E0-41 F5-23 AD-01 00-00 00-42 F5-23 AD-01 00-00 00-42 F5-23 AD-01 00-00 30-4C F5-23 AD-01 00-00 50-4C F5-23 AD-01 00-00 50-4C F5-23 AD-01 00-00 60-4F F5-23 AD-01 00-00 80-4F F5-23 AD-01 00-00 80-4F F5-23 AD-01 00-00 40-48 F5-23 AD-01 00-00 60-48 F5-23 AD-01 00-00 60-48 F5-23 AD-01 00-00>, 1, 1, 0) (0 ms)
Wed, 21 Jan 2026 03:52:32 GMT [ RUN      ] Smoke/GPU_OpTensorFwdBiasTest_FP32.PortTest/44
Wed, 21 Jan 2026 04:40:34 GMT Error: The action 'Test' has timed out after 60 minutes.
```
